### PR TITLE
Use new types library.

### PR DESCRIPTION
**Pull request by cmeiklejohn**
No more Riak DT!

### CODE REVIEW

**Comment by cmeiklejohn** [comment]
The structure for deltas has changed, so anywhere where "{delta, ..." is used, is going to be problematic.

```
ChristophersMBP:lasp cmeiklejohn$ grep -r "{delta, " src test
src/lasp_advertisement_counter.erl:                                      {ok, {_, _, _, Counter1}} = lasp:bind(Ad, {delta, Delta}),
src/lasp_advertisement_counter.erl:            {ok, {delta, Delta}} = lasp_type:update(CounterType, increment, Id, MergedCounter),
src/lasp_core.erl:-spec read_delta(id(), value(), store()) -> {ok, {delta, var()}}.
src/lasp_core.erl:            WDT = lists:append(Object#dv.waiting_delta_threads, [{delta, Self}]),
src/lasp_core.erl:bind(Id, {delta, Value}, MetadataFun, Store) ->
src/lasp_core.erl:                                                 {ok, {delta, {Id, Type, Metadata, Value}}}),
src/lasp_core.erl:                        {NewObject, {ok, {Id, Type, Metadata, {delta, Merged}}}};
src/lasp_core.erl:                        {Object, {ok, {Id, Type, Metadata, {delta, Merged}}}}
src/lasp_core.erl:             ({delta, {_, T, _, V}}) ->
src/lasp_core.erl:                  {ok, _} = BindFun(AccId, {delta, AccValue}, Store)
src/lasp_core.erl:reply_to_all([{delta, From}|T], StillWaiting, {ok, {delta, Value}}=Result) ->
src/lasp_core.erl:            gen_server:reply({Address, Ref}, {ok, {delta, Value}});
src/lasp_core.erl:                               {ok, undefined, {delta, Value}});
src/lasp_core.erl:            gen_server:reply({Address, Ref}, {ok, {delta, Value}});
src/lasp_plumtree_broadcast_distribution_backend.erl:                              {delta, MergedState} ->
src/lasp_plumtree_broadcast_distribution_backend.erl:                              {delta, MergedState} ->
src/lasp_plumtree_broadcast_distribution_backend.erl:    {delta, Deltas}.
src/lasp_process.erl:                    {delta, {_, _, _, V}} ->
src/lasp_process.erl:                        {delta, V}
```

File: include/lasp.hrl [hunk 1]
```diff
@@ -88,7 +82,7 @@
 -type var() :: {id(), type(), metadata(), value()}.
 
 %% @doc Only CRDTs are able to be processed.
--type crdt() :: riak_dt:crdt().
+-type crdt() :: types:crdt().
```
Comment (vitorenesduarte) [inline include/lasp.hrl]: Maybe

```erlang
-type crdt() :: type:crdt().
```

Comment (cmeiklejohn) [inline include/lasp.hrl]: Yes, you are right.

**Comment by cmeiklejohn** [comment]
I ported ad84eeb95e7bd884e111201963b2134b37db584b over, but I'm not convinced this is correct behavior.  Aren't we supposed to broadcast the delta immediately?  @vitorenesduarte @junghunyoo

**Comment by vitorenesduarte** [comment]
@cmeiklejohn In the delta paper you don't broadcast immediately. You gossip delta-groups from time to time.

**Comment by cmeiklejohn** [comment]
@vitorenesduarte OK, so that means 8f4c51b should be better, right?

**Comment by vitorenesduarte** [comment]
@cmeiklejohn Yes,it makes sense.
I just ported that to the other branch,

**Comment by cmeiklejohn** [comment]
What branch?

**Comment by vitorenesduarte** [comment]
[types_2](https://github.com/lasp-lang/lasp/tree/types_2).
Today I have classes, but tomorrow I'll try to finish those changes.

**Comment by cmeiklejohn** [comment]
What are the differences between types_2 and this branch?

**Comment by vitorenesduarte** [comment]
types_2 is a fork of this one, but I'm changing __delta_mode__ to __mode__ and the possible values instead of true and false are state_based, delta_based, pure_op_based.
Makes sense?

**Comment by cmeiklejohn** [comment]
Sounds good.  What about the delta related functions that are broken in
types?

**Comment by vitorenesduarte** [comment]
Still broken. 
I'll try to solve it tomorrow (still figuring out the flow of things in Lasp).

**Comment by cmeiklejohn** [comment]
Sounds good.  I think we can remove *a lot* of it.  Ping me on slack later and we can chat about it.

**Pull request merged**